pytorch_lightning/profiler/profilers.py

```diff
@@ -90,13 +108,25 @@ def describe(self) -> None:
         """Logs a profile report after the conclusion of the training run."""
         for write in self.write_streams:
             write(self.summary())
+        if self.output_file:
+            self.output_file.flush()
 
-    @abstractmethod
-    def summary(self) -> str:
-        """Create profiler summary in text format."""
-
-    def on_train_start(self, local_rank: Optional[int] = None):
-        self.local_rank = local_rank
+    def stats_to_str(self, stats: dict) -> str:
+        output = ["Profiler Report"]
+        for action, value in stats.items():
+            header = f"Profile stats for: {action}"
+            if getattr(self, "local_rank", None) is not None:
+                header += f" rank: {self.local_rank}"
+            output.append(header)
+            output.append(value)
+        return os.linesep.join(output)
+
+    def __del__(self) -> None:
+        """Close profiler's stream."""
+        try:
+            self.output_file.close()
+        except AttributeError:
+            pass
```

Comment:
- I've seen doing this in `__del__ ` sometimes runs into lifecycle issues with filesystem clients for cloud storage (hence the motivation for #6370)
- the try/catch for accessing `output_file` is very hacky :(

Reply:
Agree, removed.